File: android/app/src/main/java/com/metaone/metaone_sdk_demo/api/ApiService.java
package com.metaone.metaone_sdk_demo.api;

import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

import com.metaone.metaone_sdk_demo.api.response.SampleSSOLoginResponse;


public interface ApiService {
    @POST("https://ws-test.aag.ventures/sso/login")
    Call<SampleSSOLoginResponse> sampleSsoLogin(@Body RequestBody request);
}

File: react-native/src/api/ApiService.ts
import axios from 'axios';

type RequestType = {
  email: string;
  password: string;
};

type ResponseType = {
  token: string;
};

export const sampleSSOLogin = (
  request: RequestType,
  onSuccess: (data: ResponseType) => void,
  onFailure: (error: any) => void,
): void => {
  const url = 'https://ws-test.aag.ventures/sso/login';
  axios
    .post<ResponseType>(url, request, {
      headers: {
        'Content-Type': 'application/json',
      },
    })
    .then(response => {
      onSuccess(response.data);
    })
    .catch(error => {
      if (error.response) {
        console.log(error.response.data);
        console.log(error.response.status);
        console.log(error.response.headers);
      } else if (error.request) {
        console.log(error.request);
        console.log(error);
      } else {
        console.log('Error', error.message);
        console.log('Error', error);
      }
      console.log(error.config);
      onFailure(error);
    });
};
